rflx/identifier.py

```diff
@@ -1,26 +1,39 @@
-from typing import Sequence, Union
+from typing import Optional, Sequence, Union
 
 from rflx.common import generic_repr
 from rflx.contract import invariant
+from rflx.error import Location
 
 
 @invariant(lambda self: all(" " not in part for part in self.parts), "no whitespace in identifier")
 @invariant(lambda self: "" not in self.parts, "no empty part in identifier")
 class ID:
-    def __init__(self, identifier: Union[str, Sequence[str], "ID"] = None) -> None:
+    def __init__(
+        self, identifier: Union[str, Sequence[str], "ID"] = None, location: Location = None
+    ) -> None:
         self.parts: Sequence[str]
-        if isinstance(identifier, str):
+        self.location = location
+
+        if identifier is None:
+            self.parts = []
```

Comment:
I agree that we shouldn't open up this can of worms. I moved the optional/empty names into the AST in eed3dd896de9080dff9593df610440e83b4f894b, which should be less problematic.